src/quicklist.c

```diff
@@ -813,6 +816,11 @@ int quicklistReplaceAtIndex(quicklist *quicklist, long index, void *data,
         entry.node->entry = ziplistReplace(entry.node->entry, entry.zi, data, sz);
         quicklistNodeUpdateSz(entry.node);
         quicklistCompress(quicklist, entry.node);
+        /* quicklistIndex provides an uncompressed node */
+        entry.node->entry = ziplistReplace(entry.node->entry, entry.zi, data, sz);
+        quicklistNodeUpdateSz(entry.node);
+        quicklistCompress(quicklist, entry.node);
+        quicklistReplaceEntry(quicklist, &entry, data, sz);
```

Comment:
It looks like a wrong conflict was resolved.

Reply:
@perryitay now the conflict is properly resolved (diff-wise), but i think there's a bug.
there's a reason these few lines got extracted from `quicklistReplaceAtIndex` to `quicklistReplaceEntry`.
since now `quicklistReplaceEntry` is an API of it's own (used in t_list.c), all your plain node handling code needs to move to the inner method.